line_profiler/_line_profiler.pyx

```diff
@@ -1441,12 +1446,12 @@ pystate.h#L16
         if manager_._set_frame_local_trace:
             set_local_trace(<PyObject *>manager_, py_frame)
     elif what == PyTrace_LINE or what == PyTrace_RETURN:
-        # Normally we'd need to DECREF the return from
-        # `PyFrame_GetCode()`, but Cython does that for us
+        code = <PyObject *>PyFrame_GetCode(py_frame)
         inner_trace_callback((what == PyTrace_LINE),
                              manager_.active_instances,
-                             <object>PyFrame_GetCode(py_frame),
+                             <object>code,
                              PyFrame_GetLineNumber(py_frame))
+        Py_XDECREF(code)
```

Comment:
Apparently it does not do that for us? 😄

Reply:
I didn't notice this PR earlier, but I believe your guess is correct! I know that around 4.1.0 that comment about not needing DECREF was correct, as I had checked it with my intuition and one of the creators of memray. However, casting to an object would indeed interfere with the auto-DECREFing, since at that point it gets turned back into a Python object without Cython doing additional checks for whether DECREF is needed.